File: Sum of Left Leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from queue import Queue

class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        qu = Queue()
        qu.put((root, False))
        sum = 0

        while not qu.empty():
            node, left = qu.get()

            if left and not node.left and not node.right:
                sum += node.val
            if node.left:
                qu.put((node.left, True))
            if node.right:
                qu.put((node.right, False))
                
        return sum

File: Daily challenge/Sum of Left Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution 
{
public:
    int sumOfLeftLeaves(TreeNode* root) 
    {
        queue<pair<TreeNode*, bool>> qu;
        qu.push({root, false});
        int sum = 0;

        while(!qu.empty())
        {
            auto [node, left] = qu.front();
            qu.pop();

            if(left && !node->left && !node->right)
            {
                sum += node->val;
            }
            if(node->left)
            {
                qu.push({node->left, true});
            }
            if(node->right)
            {
                qu.push({node->right, false});
            }
        }
        return sum;
    }
};